packages/account-v2/src/modules/Onfido/OnfidoContainer.tsx

```diff
@@ -0,0 +1,115 @@
+    const history = useHistory();
+
+    const {
+        data: { hasSubmitted, onfidoContainerId, onfidoRef },
+        isOnfidoInitialized,
+        isServiceTokenLoading,
+        onfidoInitializationError,
+        serviceTokenError,
+    } = useOnfido(country);
+
+    useEffect(() => {
+        if (hasSubmitted) {
+            onOnfidoSubmit?.();
+            onfidoRef?.current?.safeTearDown();
+        }
+    }, [hasSubmitted, onfidoRef, onOnfidoSubmit]);
+
+    useEffect(() => {
+        let transitionEndTimeout: ReturnType<typeof setTimeout>;
+        if (isOnfidoEnabled) {
+            transitionEndTimeout = setTimeout(() => setTransitionEnd(true), 350);
+        }
+        return () => {
+            if (transitionEndTimeout) {
+                clearTimeout(transitionEndTimeout);
+            }
+        };
+    }, [isOnfidoEnabled]);
+
+    const showMissingPersonalDetails = ['MissingPersonalDetails', 'InvalidPostalCode'].includes(
+        serviceTokenError?.error.code ?? ''
+    );
+    const showErrorMessage = onfidoInitializationError?.message || serviceTokenError?.error?.message;
+
+    if (isServiceTokenLoading) {
+        return <Loader />;
+    } else if (showMissingPersonalDetails) {
+        const invalidPostalCode = serviceTokenError?.error.code === 'InvalidPostalCode';
+        const missingDetailTitle = invalidPostalCode
+            ? 'Your postal code is invalid'
+            : 'Your personal details are missing';
+        const missingDetailMessage = invalidPostalCode
+            ? 'Please check and update your postal code before submitting proof of identity.'
+            : 'Please complete your personal details before you verify your identity.';
+        const missingDetailButtonText = invalidPostalCode ? 'Update postal code' : 'Go to personal details';
+
+        return (
+            <IconWithMessage
+                actionButton={
+                    <Button
+                        onClick={() =>
+                            history.push(
+                                `/account/personal-details?from=proof_of_identity${
+                                    invalidPostalCode ? '#address_postcode' : ''
+                                }`
+                            )
+                        }
+                    >
+                        {missingDetailButtonText}
+                    </Button>
+                }
+                icon={<IcAccountMissingDetails width={128} />}
+                title={missingDetailTitle}
+            >
+                <Text>{missingDetailMessage}</Text>
+            </IconWithMessage>
+        );
+    } else if (showErrorMessage) {
+        return <ErrorMessage message={showErrorMessage} />;
+    }
+
+    return (
+        <div className={qtMerge('flex flex-col items-center gap-800 p-800 lg:p-50')}>
+            <div
+                className={clsx(
+                    '[transition:transform_0.35s_ease,_opacity_0.35s_linear]origin-top opacity-1300 p-800 max-w-[670px]',
```

Comment:
Is there a better way to size the components rather than restricting the width?

Reply:
removed. I don't think we need it here, it should be handled by child's width.